x/authz/client/cli/query.go

```diff
@@ -0,0 +1,138 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/cosmos/cosmos-sdk/x/authz/types"
+)
+
+// GetQueryCmd returns the cli query commands for this module
+func GetQueryCmd() *cobra.Command {
+	authorizationQueryCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "Querying commands for the msg authorization module",
+		Long:                       "",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	authorizationQueryCmd.AddCommand(
+		GetCmdQueryAuthorization(),
+		GetCmdQueryAuthorizations(),
+	)
+
+	return authorizationQueryCmd
+}
+
+// GetCmdQueryAuthorizations implements the query authorizations command.
+func GetCmdQueryAuthorizations() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "authorizations [granter-addr] [grantee-addr]",
+		Args:  cobra.ExactArgs(2),
+		Short: "query list of authorizations for a granter-grantee pair",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Query list of authorizations for a granter-grantee pair:
+Example:
+$ %s query %s authorizations cosmos1skj.. cosmos1skjwj..
+`, version.AppName, types.ModuleName),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			granterAddr, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			granteeAddr, err := sdk.AccAddressFromBech32(args[1])
+			if err != nil {
+				return err
+			}
+
+			pageReq, err := client.ReadPageRequest(cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			res, err := queryClient.Authorizations(
+				context.Background(),
+				&types.QueryAuthorizationsRequest{
+					Granter:    granterAddr.String(),
+					Grantee:    granteeAddr.String(),
+					Pagination: pageReq,
+				},
+			)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+	flags.AddQueryFlagsToCmd(cmd)
+	flags.AddPaginationFlagsToCmd(cmd, "authorizations")
+	return cmd
+}
+
+// GetCmdQueryAuthorization implements the query authorization command.
+func GetCmdQueryAuthorization() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "authorization [granter-addr] [grantee-addr] [msg-type]",
+		Args:  cobra.ExactArgs(3),
+		Short: "query authorization for a granter-grantee pair",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Query authorization for a granter-grantee pair that matches the given msg-type:
+Example:
+$ %s query %s authorization cosmos1skjw.. cosmos1skjwj.. %s
+`, version.AppName, types.ModuleName, types.SendAuthorization{}.MethodName()),
```

Comment:
is it possible to make it easier for the user - passing method names as `/cosmos.bank.v1beta1.Msg/Send` would make it complex for generic authorizations

Reply:
I would suggest to keep the experience consistent for all the clients. Other apps who are building on top of SDK will have custom msgs, so we cannot actually simplify this and it would be inconsistent as well. wdyt @sahith-narahari ?

Reply:
yeah, don't see much options here. Should be fine with the validations in place